test/e2e_test.go

```diff
@@ -255,11 +260,11 @@ var _ = Describe("e2e", func() {
 				EndorsementPlugin:   "",
 				ValidationPlugin:    "",
 				Sequence:            1,
-				ValidationParameter: nil,
+				ValidationParameter: AppPolicydata,
```

Comment:
I wonder if it would be helpful for users to change the ValidationParameter type to `*peer.ApplicationPolicy` so the API guides them to the information required for this field, and avoids them needing to marshal the protobuf themselves. All our internal points of usage for the Definition would then just need to do that marshal step when constructing transaction payloads from the Definition.

Reply:
I suppose it will be in another PR, as not all operators(for example query) need to do right?
if only chaincode Approve need this, I will cover in this pr. otherwise it will be in another one.

Reply:
ref to https://hyperledger-fabric.readthedocs.io/en/latest/commands/peerlifecycle.html I will update this PR with approve, commit and checkcommitreadiness